x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/EventEmitterService.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.analytics;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.MarkerManager;
+
+import org.elasticsearch.ResourceNotFoundException;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.client.internal.OriginSettingClient;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.ReleasableBytesStreamOutput;
+
+import org.elasticsearch.common.util.BigArrays;
+
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+
+
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime;
+
+import static org.elasticsearch.xpack.core.ClientHelper.ENT_SEARCH_ORIGIN;
+
+public class EventEmitterService {
+    private static final Logger logger = LogManager.getLogger(EventEmitterService.class);
+    private static final Marker AUDIT_MARKER = MarkerManager.getMarker("org.elasticsearch.xpack.entsearch.analytics");
+    public static final String ANALYTICS_DATASTREAM_PATTERN = ".events-";
```

Comment:
What is this pattern? We shouldn't use such a generic name. We should also avoid using `.` as a prefix, this is reserved for system index. I thought that @afoucret added the index templates in the registry with a prefix like `behavioral_analytics-events-`?

Reply:
This is going to be replaced with @afoucret recent changes

Reply:
ok cool, sorry for the early comment then. That scared me a bit to be honest ;)

Reply:
Sorry, didn't mean to scary you 😅 It's just a draft version, I'm gonna polish it tomorrow and on Friday

Reply:
np, that's a great start btw!